BootloaderCommonPkg/Library/IppCryptoLib/auth/pcpsha256ca.c

```diff
@@ -167,14 +167,20 @@ void UpdateSHA256Compact(void* uniHash, const Ipp8u* mblk, int mlen, const void*
 
 void UpdateSHA256(void* pHash, const Ipp8u* pMsg, int msgLen, const void* pParam)
 {
-#if FixedPcdGetBool (PcdCryptoShaNiEnabled)
-    UpdateSHA256Ni(pHash, pMsg, msgLen, pParam);
+#if defined(_SLIMBOOT_OPT)
+
+  if (FixedPcdGet32 (PcdCryptoShaOptEnabled) & IPP_CRYPTO_SHA256_NI) {
+      UpdateSHA256Ni(pHash, pMsg, msgLen, pParam);
+   } else if (FixedPcdGet32 (PcdCryptoShaOptEnabled) & IPP_CRYPTO_SHA256_V8) {
```

Comment:
You should keep V8 as a default for SHA256.  So that if no bit is set, V8 will be used.

Reply:
In TGL, Ni optimization is for SHA256 is 5 times better than V8 optimization.
I would set Ni as default value in BuildLoader.ENABLE_CRYPTO_SHA_OPT

Reply:
Well, more generic means supported by most of the platform.  NI has very limited platform support.